Core/include/Acts/Visualization/IVisualization.hpp

```diff
@@ -104,13 +112,35 @@ class IVisualization {
     Vector3D bd = b.template cast<double>();
     line(ad, bd, color);
   }
+
+  /// Helper: check for extension
+  ///
+  /// @note this is a placeholder for std::filesystem::has_extension
+  /// which needs special linking until gcc9
+  /// @param path the path to be checked
+  bool hasExtension(const std::string& path) const {
+    return (path.find(".") != std::string::npos);
+  }
```

Comment:
Can you move the implementations of these methods into a cpp file? I didn't do it initially I think but I don't think there was a specific reason. 

Also, can you make these helper functions protected? I don't think you need them outside of the derived classes, do you?

Reply:
Sure I can.

Reply:
Done, moved to `.cpp` and made `protected`.